bgv/decryptor.go

```diff
@@ -1,55 +0,0 @@
-package bgv
-
-import (
-	"github.com/tuneinsight/lattigo/v4/ring"
-	"github.com/tuneinsight/lattigo/v4/rlwe"
-)
-
-// Decryptor is an interface wrapping a rlwe.Decryptor.
-type Decryptor interface {
-	DecryptNew(ciphertext *Ciphertext) (plaintext *Plaintext)
-	Decrypt(ciphertext *Ciphertext, plaintext *Plaintext)
-	ShallowCopy() Decryptor
-	WithKey(sk *rlwe.SecretKey) Decryptor
-}
-
-type decryptor struct {
-	rlwe.Decryptor
-	params Parameters
-	buffQ  *ring.Poly
-}
-
-// NewDecryptor instantiates a Decryptor for the BGV scheme.
-func NewDecryptor(params Parameters, sk *rlwe.SecretKey) Decryptor {
```

Comment:
We should keep this method (with rlwe.Decryptor output).

Reply:
Done